### ISSUE TITLE
Stage simple should not add to Backbone.History

### ISSUE THREAD

**Issue by mikeblum** [tsgrp/hpi#632]
This issue is because of how our Backbone routers are designed in HPI. see these issues:

https://github.com/jashkenas/backbone/issues/3183
https://github.com/jashkenas/backbone/issues/3182

The gist of the issue is that Backbone.History doesn't support having multiple routers in memory. The solution is to either have a single router that imports and extends other routes on the routes: attribute or (as things stand now with our router configuration) to call the route directly using app.routers.{{router name}}.{{route}}

Basically everywhere we call {replace: true} will need this change:

- [x] View Versions
- [x] Saved Search Dashlets (All Dashlets)

**Comment by gsteimer** [comment]
I'm almost positive this is because the stage simple route adds itself to the history.  You can also get this to happen:

1. Click on a link in a saved search dashlet (ex: AW Form).
1. It loads in the stage
1. Clicking back will just load the stage again

**Comment by mikeblum** [comment]
Revision: 11949
Author: mblum
Date: 2015-03-27 10:14:03.107538
SAVED SEARCH DASHLET: doesn't add StageSimple to browser history but does for Stage. Fixes infinite back button to the Stage issue fr this route.

M	/trunk/code/app/modules/dashboard/savedsearchdashlet.js

**Comment by gsteimer** [comment]
@mikeblum - can you verify this on edge1/2?  Stage simple still seems to be in the history for me.  Or - did your fix only handle the case in your original post?  I tested from dashboard, and it's still not working.

**Comment by gsteimer** [comment]
Looking at your commit again - looks like you updated saved search dashlet JS.... I'm trying that exact scenario on edge1/2...

**Comment by b-k-white** [comment]
Have this fixed locally, waiting for a CR tomorrow morning

**Comment by b-k-white** [comment]
Fixed issue with backbone.history. This fix solves the back button problem that stageSimple would create. Solved it by calling the router directly so that the backbone.history method isn't called twice creating the loop. This is not a long term fix, because we still have multiple routers which is not an ideal situation.

CR - Zaq
HPI - 12662

**Comment by b-k-white** [comment]
Not sure if we want to close or leave this open based on the fact that we still use multiple routers.

**Comment by gsteimer** [comment]
@b-k-white - I don't think this is fixed.  See the screencam below from edge2.  Clicking on recent folders, which routes through stageSimple, still has the back button problem.

![stage-simple-route](https://cloud.githubusercontent.com/assets/1377228/8598519/b1be91b2-2620-11e5-8c2f-149e0699911f.gif)

**Comment by b-k-white** [comment]
Thought I found all of the calls to stageSimple, but I seemed to have missed recent objects.

**Comment by b-k-white** [comment]
Recent objects fix in
HPI - 12696
CR - Zaq
@gsteimer Should this issue be closed if the recent objects issue is solved? We might want to keep it open to remind us that we should be reducing our routers into one.

**Comment by gsteimer** [comment]
@b-k-white @zkeller - We can close this when StageSimple is no longer added to the back button history everywhere in HPI.  

It sounds to me that the whole reduce routers to one thing should be a separate issue.  Do you guys agree?

**Comment by gsteimer** [comment]
@another place this isn't working - Inbox (normal Inbox, not Wizard Inbox, which works fine).

![inbox-stage-simple](https://cloud.githubusercontent.com/assets/1377228/8884389/050b162c-321c-11e5-89f7-e2f0d2845b06.gif)

@b-k-white is there a way to factor this out into stage simple itself (or somewhere else) so that we don't have to keep remembering to do this everywhere we use it?  I'm thinking that whoever does #749 shouldn't have to worry about this if we fix it in the right spot...

**Comment by nquillin91** [comment]
Tableview currently doesn't handle StageSimple calls appropriately due to the fact that all the links via Search are hrefs. Most likely extensive modification will be required to utilize onClick events versus relying on the hrefs.
